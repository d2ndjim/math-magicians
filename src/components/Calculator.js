import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Calculator() {
  const [state, setState] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setState((state) => ({
      ...state,
      ...calculate(state, e.target.name),
    }));
  };

  const { total, next, operation } = state;

  return (
    <>
      <h4 className="calculator-text">Lets do some Math!</h4>
      <section>
        <div className="btn-rows">
          <div id="result" className="grid-item">
            {total}
            {operation}
            {next}
          </div>

          <button type="button" className="grid-item" name="AC" onClick={handleClick}>AC </button>
          <button type="button" className="grid-item" name="+/-" onClick={handleClick}> +/- </button>
          <button type="button" className="grid-item" name="%" onClick={handleClick}> % </button>
          <button type="button" className="grid-item orange" name="รท" onClick={handleClick}> รท </button>
          <button type="button" className="grid-item" name="7" onClick={handleClick}> 7 </button>
          <button type="button" className="grid-item" name="8" onClick={handleClick}> 8 </button>
          <button type="button" className="grid-item" name="9" onClick={handleClick}> 9 </button>
          <button type="button" className="grid-item orange" name="x" onClick={handleClick}> x </button>
          <button type="button" className="grid-item" name="4" onClick={handleClick}> 4 </button>
          <button type="button" className="grid-item" name="5" onClick={handleClick}> 5 </button>
          <button type="button" className="grid-item" name="6" onClick={handleClick}> 6 </button>
          <button type="button" className="grid-item orange" name="-" onClick={handleClick}> - </button>
          <button type="button" className="grid-item" name="1" onClick={handleClick}> 1 </button>
          <button type="button" className="grid-item" name="2" onClick={handleClick}> 2 </button>
          <button type="button" className="grid-item" name="3" onClick={handleClick}> 3 </button>
          <button type="button" className="grid-item orange" name="+" onClick={handleClick}> + </button>
          <button type="button" className="grid-item zero" name="0" onClick={handleClick}> 0 </button>
          <button type="button" className="grid-item" name="." onClick={handleClick}> . </button>
          <button type="button" className="grid-item orange" name="=" onClick={handleClick}> = </button>
        </div>
      </section>
    </>

  );
}

export default Calculator;
