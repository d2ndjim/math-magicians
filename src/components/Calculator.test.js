import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

test('matches home snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
