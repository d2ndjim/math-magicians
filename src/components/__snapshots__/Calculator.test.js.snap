// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches home snapshot 1`] = `
Array [
  <h4
    className="calculator-text"
  >
    Lets do some Math!
  </h4>,
  <section>
    <div
      className="btn-rows"
    >
      <div
        className="grid-item"
        id="result"
      >
        0
      </div>
      <button
        className="grid-item"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC 
      </button>
      <button
        className="grid-item"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
         +/- 
      </button>
      <button
        className="grid-item"
        name="%"
        onClick={[Function]}
        type="button"
      >
         % 
      </button>
      <button
        className="grid-item orange"
        name="รท"
        onClick={[Function]}
        type="button"
      >
         รท 
      </button>
      <button
        className="grid-item"
        name="7"
        onClick={[Function]}
        type="button"
      >
         7 
      </button>
      <button
        className="grid-item"
        name="8"
        onClick={[Function]}
        type="button"
      >
         8 
      </button>
      <button
        className="grid-item"
        name="9"
        onClick={[Function]}
        type="button"
      >
         9 
      </button>
      <button
        className="grid-item orange"
        name="x"
        onClick={[Function]}
        type="button"
      >
         x 
      </button>
      <button
        className="grid-item"
        name="4"
        onClick={[Function]}
        type="button"
      >
         4 
      </button>
      <button
        className="grid-item"
        name="5"
        onClick={[Function]}
        type="button"
      >
         5 
      </button>
      <button
        className="grid-item"
        name="6"
        onClick={[Function]}
        type="button"
      >
         6 
      </button>
      <button
        className="grid-item orange"
        name="-"
        onClick={[Function]}
        type="button"
      >
         - 
      </button>
      <button
        className="grid-item"
        name="1"
        onClick={[Function]}
        type="button"
      >
         1 
      </button>
      <button
        className="grid-item"
        name="2"
        onClick={[Function]}
        type="button"
      >
         2 
      </button>
      <button
        className="grid-item"
        name="3"
        onClick={[Function]}
        type="button"
      >
         3 
      </button>
      <button
        className="grid-item orange"
        name="+"
        onClick={[Function]}
        type="button"
      >
         + 
      </button>
      <button
        className="grid-item zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
         0 
      </button>
      <button
        className="grid-item"
        name="."
        onClick={[Function]}
        type="button"
      >
         . 
      </button>
      <button
        className="grid-item orange"
        name="="
        onClick={[Function]}
        type="button"
      >
         = 
      </button>
    </div>
  </section>,
]
`;
